pkg/services/folder/folderimpl/folder.go

```diff
@@ -1241,11 +1247,10 @@ func (s *Service) nestedFolderDelete(ctx context.Context, cmd *folder.DeleteFold
 		descendantUIDs = append(descendantUIDs, f.UID)
 	}
 	s.log.InfoContext(ctx, "deleting folder and its descendants", "org_id", cmd.OrgID, "uid", cmd.UID)
-	toDelete := append(descendantUIDs, cmd.UID)
-	err = s.store.Delete(ctx, toDelete, cmd.OrgID)
+
+	err = s.store.Delete(ctx, descendantUIDs, cmd.OrgID)
 	if err != nil {
-		s.log.InfoContext(ctx, "failed deleting folder", "org_id", cmd.OrgID, "uid", cmd.UID, "err", err)
-		return descendantUIDs, err
+		s.log.InfoContext(ctx, "failed deleting descendants", "org_id", cmd.OrgID, "parent_uid", cmd.UID, "err", err)
```

Comment:
This change is a blocker for this PR, so I have it in the PR, but I'm having it reviewed here: https://github.com/grafana/grafana/pull/99399. Will have that merged in and merge in main here before merging this PR in